tracsubtickets/api.py

```diff
@@ -69,7 +69,8 @@ def __init__(self):
     # IEnvironmentSetupParticipant methods
     def environment_created(self):
         self.found_db_version = 0
-        self.upgrade_environment(self.env.get_db_cnx())
+        with self.env.db_transaction as db:
+            self.upgrade_environment(db)
```

Comment:
For compatibility with future versions of Trac (1.3.1 and later):
1. Change signature of `upgrade_environment` to `def upgrade_environment(self, db=None):`
2. Use `self.env.db_transaction` in `upgrade_environment` to get `db` connection.
3. Remove context manager in `environment_created` and just call `self.upgrade_environment()`.